### PR TITLE
Added examples for usage in a middleware

### PR DESCRIPTION
**Pull request by kjellkongsvik**

### CODE REVIEW

File: examples/middleware/src/auth.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    pub jwks: Jwks,
+}
+
+impl FromRef<AppState> for Jwks {
+    fn from_ref(state: &AppState) -> Self {
+        state.jwks.clone()
+    }
+}
+
+pub async fn my_middleware<B>(
```
Comment (cdriehuys) [inline examples/middleware/src/auth.rs]: Could you add a docstring to this middleware? I think it's especially important for examples to be very clear about what they're doing. Something stating that it's a middleware that verifies a valid token is provided, but does not utilize any claims.

**Review by cdriehuys — CHANGES_REQUESTED** [review]
I left one suggestion, but also have two more questions:
1. Are examples built as part of the CI build?
2. I fixed some broken links in the main branch. Can you rebase on top of those changes?

File: examples/middleware/src/auth.rs (hunk 1 above)
Comment (kjellkongsvik) [reply examples/middleware/src/auth.rs]: Thanks, also rename to `validate_token`

**Comment by kjellkongsvik** [comment]
1. The examples are checked in the CI, not built. Should be enough?
2. Yes, rebased.

**Review by cdriehuys — APPROVED** [review]

**Pull request merged**